### PR TITLE
fix(branch):add login endpoint #184651331

### PR DESCRIPTION
**Pull request by bstwagir**
#### What does this PR do?
Fix a bug found in PR #32: #184651331 Add login API endpoint.
Using an auto-generated testToken variable that refreshes itself every time tests run instead of a hard-coded number in the login.test file.

#### Description of Task to be completed?

Have the following endpoint working
POST /api/login

#### How should this be manually tested?

============================ PRE-REQUIRED: =================================
====> .env

USER_SECRET_KEY= YOUR JWT SECRET KEY 
NODE_ENV= ENV YOU WANT YOU USE FROM YOUR CONFIG
DB_USERNAME= YOUR POSTGRES USERNAME
DB_PASSWORD= YOUR POSTGRES PASSWORD
DB_DEVELOPMENTDB=YOUR DATABASE NAME
DB_HOST=YOUR HOSTNAME

====> Needed
User in the database

====>
- Clone the repository 
- Navigate to the project 
- In the terminal type: npm install to install the required dependencies.
* RUN npm start
* Using Postman test the endpoint above

#### What are the relevant pivotal tracker/Trello stories?
#184651331

### CODE REVIEW

**Review by Raphaela664 — DISMISSED** [review]

**Review by nshutip — COMMENTED** [review]
The changes are well implemented but the PR name and description are do not meet the engineering playbook standards, and are therefore confusing.

File: src/controllers/login.controller.js [hunk 1]
```diff
@@ -31,7 +31,7 @@ export class PublicController {
         { id: doesExist._id, email: doesExist.email },
         process.env.USER_SECRET_KEY,
 
-        { expiresIn: 14400, }
+        { expiresIn: '7d', }
```
Comment (didasmbalanya) [inline src/controllers/login.controller.js]: Set this a configurable. i.e as an env variable but with a default value

File: src/tests/app.test.js [hunk 2]
```diff
@@ -93,7 +93,7 @@ describe('Build Tests', () => {
     // expect(res.body).to.have.property('message', 'this account does not exist');
   });
   it('It should grant access to a user with a valid token ', async () => {
-    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImFiY0BnbWFpbC5jb20iLCJpYXQiOjE2ODA0ODc5ODMsImV4cCI6MTY4MDUwMjM4M30.udpfqgyx72GI-d8pLi-ik2BszbAn-tTRLExlr0KBoaA';
+    const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImFiY0BnbWFpbC5jb20iLCJpYXQiOjE2ODA1NzQ5OTQsImV4cCI6MTY4MTE3OTc5NH0.843FO-wtu3P-rVAG5VIGFB0Ph3VT_dkCncMFrY0sxOI';
```
Comment (didasmbalanya) [inline src/tests/app.test.js]: Why not generate a token here? This will fail the tests in the future after the token expires

**Review by didasmbalanya — DISMISSED** [review]

File: src/tests/app.test.js
Comment (heshima18) [inline src/tests/app.test.js:1]: make auto generated tokens to prevent the tests from failing in  the upcoming time

**Review by heshima18 — CHANGES_REQUESTED** [review]
update the token concept

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]

**Review by kagame8732 — APPROVED** [review]
Well designed

**Review by JulesNIYOMWUNGERI — APPROVED** [review]

**Review by ruberwa — CHANGES_REQUESTED** [review]
Functionalities ok, but something is missing

Most cases are covered but this: 

When End-User's login fails, Supposed to see clear messages of what is wrong like 
[x] Your Email is invalid or something else just to let the user know that is giving wrong inputs: When something else, not email is given

**Pull request merged**